lua/ttt2/extensions/table.lua

```diff
@@ -288,6 +288,35 @@ function table.ExtractRandomEntry(tbl, filterFn)
 end
 
 ---
+-- Scans the given table for the subtable with the most entries. This then
+-- is returned and deleted from the source table. If there are multiple subtables with
+-- the same size, the first found will be returned.
+-- @note This function does not work recursivly. Therefore the size of the sub-subtables
+-- has no effect on the size of the subtables.
+-- @param table The table whose biggest subtable should be found.
+-- @return[default={}] table The biggest subtable
+-- @return[default=0] number|string The index where the biggest subtable was found
+-- @realm shared
+function table.GetAndRemoveBiggestSubTable(tbl)
+	local subTbl = {}
+	local subIdx = 0
+
+	for i, t in pairs(tbl) do
+		if #t < #subTbl then continue end
+
+		subTbl = t
+		subIdx = i
+	end
+
+	if isnumber(subItx) then
+		table.remove(tbl, subIdx)
```

Comment:
I know what you are doing, but please add a comment or hint or warning that an number indexed table has to be interable. Otherwise, this will fail. References to the binding system